packages/hardhat-core/src/internal/core/jsonrpc/types/input/blockTag.ts

```diff
@@ -0,0 +1,42 @@
+import * as t from "io-ts";
+
+import { optional } from "../../../../util/io-ts";
+import { rpcData, rpcQuantity } from "../base-types";
+
+export const rpcNewBlockTagObjectWithNumber = t.type({
+  blockNumber: rpcQuantity,
+});
+
+export const rpcNewBlockTagObjectWithHash = t.type({
+  blockHash: rpcData,
+  requireCanonical: optional(t.boolean),
+});
+
+export const rpcBlockTagName = t.keyof({
+  earliest: null,
+  latest: null,
+  pending: null,
+});
+
+// This is the new kind of block tag as defined by https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1898.md
+export const rpcNewBlockTag = t.union([
```

Comment:
I split the new, EIP-1989 block tag type and the old one. The reason for this is that Geth doesn't accept the new one everywhere. I checked when each is used and updated the Eth module accordingly.